dotnet/src/Connectors/Connectors.AzureAIInference/Services/AzureAIInferenceChatCompletionService.cs

```diff
@@ -77,20 +108,40 @@ public AzureAIInferenceChatCompletionService(
         ChatCompletionsClient chatClient,
         ILoggerFactory? loggerFactory = null)
     {
+        var logger = loggerFactory?.CreateLogger(typeof(AzureAIInferenceChatCompletionService));
         this._core = new(
             modelId,
             chatClient,
-            loggerFactory?.CreateLogger(typeof(AzureAIInferenceChatCompletionService)));
+            logger);
+
+        var builder = new ChatClientBuilder()
+         .UseFunctionInvocation(config =>
+             config.MaximumIterationsPerRequest = MaxInflightAutoInvokes);
+
+        if (logger is not null)
+        {
+            builder = builder.UseLogging(logger);
+        }
+
+        this._chatService = builder
+            .Use(this._core.Client.AsChatClient(modelId))
+            .AsChatCompletionService();
     }
 
     /// <inheritdoc/>
     public IReadOnlyDictionary<string, object?> Attributes => this._core.Attributes;
 
     /// <inheritdoc/>
     public Task<IReadOnlyList<ChatMessageContent>> GetChatMessageContentsAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
-        => this._core.GetChatMessageContentsAsync(chatHistory, executionSettings, kernel, cancellationToken);
+        => this._chatService.GetChatMessageContentsAsync(chatHistory, executionSettings, kernel, cancellationToken);
 
     /// <inheritdoc/>
     public IAsyncEnumerable<StreamingChatMessageContent> GetStreamingChatMessageContentsAsync(ChatHistory chatHistory, PromptExecutionSettings? executionSettings = null, Kernel? kernel = null, CancellationToken cancellationToken = default)
-        => this._core.GetStreamingChatMessageContentsAsync(chatHistory, executionSettings, kernel, cancellationToken);
+        => this._chatService.GetStreamingChatMessageContentsAsync(chatHistory, executionSettings, kernel, cancellationToken);
+
+    #region Private
+
+    private const int MaxInflightAutoInvokes = 128;
```

Comment:
This constant exists in both `AzureAIInferenceChatCompletionService.cs` and `AzureAIInferenceServiceCollectionExtensions.cs`. Should we define a single constant in separate file and re-use it in multiple places?